src/browser/BrowserService.cpp

```diff
@@ -198,6 +198,64 @@ QJsonObject BrowserService::getDatabaseGroups()
     return result;
 }
 
+QJsonObject BrowserService::createNewGroup(const QString& groupName)
+{
+    auto db = getDatabase();
+    if (!db) {
+        return {};
+    }
+
+    Group* rootGroup = db->rootGroup();
+    if (!rootGroup) {
+        return {};
+    }
+
+    QJsonObject result;
+    auto group = rootGroup->findGroupByPath(groupName);
```

Comment:
My only issue here is that groupName is from user-input data. We should do some basic sanitization and normalization of slashes. Is there a function in the group edit process that checks the name of a group for disallowed characters?

Reply:
I couldn't find any functions that checks input data, even when creating new groups manually.